libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java

```diff
@@ -131,48 +131,51 @@ public SSMProvider getSsmProvider() {
         return ssmProvider;
     }
 
-    public String getFeatureSet() {
-        return featureSet;
+    public List<String> getFeatureSet() {
+        return (featureSet != null && !featureSet.isEmpty())
+                ? featureSet
+                : Collections.singletonList(null);
     }
 
-    public void setFeatureSet(String featureSet) {
-        if (featureSet == null || featureSet.isBlank()) {
+    public void setFeatureSet(List<String> featureSet) {
+        if (featureSet == null || featureSet.isEmpty()) {
             this.featureSet = null;
         } else {
-            this.featureSet = featureSet;
+            this.featureSet =
+                    featureSet.stream()
+                            .filter(fs -> (fs != null && !fs.isEmpty() && !fs.isBlank()))
+                            .toList();
         }
     }
 
     public String getEnvironmentVariable(EnvironmentVariable environmentVariable) {
         return System.getenv(environmentVariable.name());
     }
 
-    public String getSsmParameter(String path) {
-        return ssmProvider.get(path);
-    }
```

Comment:
Why did this move? It feels natural to keep overrides of the same method together

Reply:
(still applies)

Reply:
Ok sure - I think it's a bit suspicious having one override be public and one be private. I wonder if we should just inline the private one - it's only used in one place.

Reply:
While refactoring should've done inline. - Done